rita_common/src/rita_loop/fast_loop.rs

```diff
@@ -90,100 +94,90 @@ impl Handler<Tick> for RitaFastLoop {
-                            Ok(())
                         })
+                    })
+                })
+                .then(|ret| {
+                    if let Err(e) = ret {
+                        error!("Failed to watch client traffic with {:?}", e)
+                    }
+                    Ok(())
                 }),
         );
 
         // Observe the dataplane for status and problems. Tunnel GC checks for specific issues
         // (tunnels that are installed but not active) and cleans up cruft. We put these together
         // because both can fail without anything truly bad happening and we get a slight efficiency
         // bonus running them together (fewer babel socket connections per loop iteration)
-        Arbiter::spawn(TunnelManager::from_registry().send(GetNeighbors).then(
-            move |rita_neighbors| {
-                let rita_neighbors = rita_neighbors.unwrap().unwrap();
-                open_babel_stream_legacy(babel_port)
-                    .from_err()
-                    .and_then(move |stream| {
-                        start_connection_legacy(stream).and_then(move |stream| {
-                            parse_routes_legacy(stream).and_then(move |(stream, babel_routes)| {
-                                parse_neighs_legacy(stream).and_then(
-                                    move |(stream, babel_neighbors)| {
-                                        parse_interfaces_legacy(stream).and_then(
-                                            move |(_stream, babel_interfaces)| {
-                                                trace!("Sending network monitor tick");
-                                                update_network_info(NetworkMonitorTick {
-                                                    babel_neighbors,
-                                                    babel_routes,
-                                                    rita_neighbors,
-                                                });
-
-                                                trace!("Sending tunnel GC");
-                                                TunnelManager::from_registry().do_send(TriggerGc {
-                                                    tunnel_timeout: TUNNEL_TIMEOUT,
-                                                    tunnel_handshake_timeout:
-                                                        TUNNEL_HANDSHAKE_TIMEOUT,
-                                                    babel_interfaces,
-                                                });
-                                                Ok(())
-                                            },
-                                        )
-                                    },
-                                )
-                            })
+        let res = tm_get_neighbors();
+        let rita_neighbors = res.unwrap();
+        Arbiter::spawn(
+            open_babel_stream_legacy(babel_port)
+                .from_err()
```

Comment:
this caught my eye, this chain of futures is used for updating network monitor and tunnel manager gc, both of which are async/await now, so we can use babel_monitor rather than legacy here.

Reply:
fixed